infrastructure/terraform/components/iam/amplify_app.tf

```diff
@@ -9,6 +9,18 @@ resource "aws_amplify_app" "main" {
   enable_branch_auto_build    = var.enable_amplify_branch_auto_build
   platform                    = "WEB_COMPUTE"
 
+  enable_basic_auth      = var.enable_amplify_basic_auth ? true : false
+  basic_auth_credentials = var.enable_amplify_basic_auth ? base64encode("${local.csi}:${aws_ssm_parameter.amplify_password[0].value}") : null
+
+  dynamic "auto_branch_creation_config" {
```

Comment:
do we need this if `enable_auto_branch_creation` is set to false already? Looked like it was related to that being there but I maybe wrong?

Reply:
yea, branches get created automatically by deploying the branch component, in doing so they inherrit this configuration. its daft